ddev/src/ddev/repo/core.py

```diff
@@ -53,10 +53,96 @@ def repo(self) -> Repository:
         return self.__repo
 
     def get(self, name: str) -> Integration:
+        if name in self.__cache:
+            return self.__cache[name]
+
         path = self.repo.path / name
         if not path.is_dir():
             raise OSError(f'Integration does not exist: {Path(self.repo.path.name, name)}')
-        elif not Integration.is_valid(path):
+
+        integration = Integration(path, self.repo.path, self.repo.config)
+        if not integration.is_valid:
             raise OSError(f'Path is not an integration nor a Python package: {Path(self.repo.path.name, name)}')
 
-        return Integration(path, self.repo.path, self.repo.config)
+        self.__cache[name] = integration
+        return integration
+
+    def iter(self) -> Iterable[Integration]:
+        for path in sorted(self.repo.path.iterdir()):
+            if path.name in self.__cache:
+                integration = self.__cache[path.name]
+                if integration.is_integration:
+                    yield integration
+            elif (integration := Integration(path, self.repo.path, self.repo.config)).is_integration:
+                self.__cache[path.name] = integration
+                yield integration
```

Comment:
Would it be better to alway add the new instance to the cache even if `is_integration` is False? Something like this:

```python
if path.name in self.__cache:
    integration = self.__cache[path.name]
else:
    integration = Integration(path, self.repo.path, self.repo.config)
    self.__cache[path.name] = integration

if integration.is_integration:
    yield integration
```

We could even have the first part (to get the integration) in another function and replicate the logic in the other `iter` functions

Reply:
Done!